### PR TITLE
 feat/LM-20-create-test-cases-for-landing-page

### PR DESCRIPTION
**Pull request by TJYumul**
## **Related Issues/Tasks**

- [LM-20](https://vgonzales5157.atlassian.net/browse/LM-20)

## PR Question
**Do you have specific questions or issues you would like to address with the reviewer?**
- none

## Description
- create test cases for landing page

## PR/Code Author Checklist
**Note:** please write "x" within the brackets if item applies e.g. [x]

[x] - I set the PR title based on the following format: feat/LM-20-create-test-cases-for-landing-page

[x] - I set the branch name based on the following format: feat/LM-20-create-test-cases-for-landing-page

[x] - I set KuroXI as the reviewer and assignee of this PR

[x] - I set the correct PR Labels

[x] - All new and existing tests passed

[x] - I did not skip/commented-out/removed any existing tests

[x] - I have performed a self-review of my code

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,16 +41,19 @@
...
+    "@testing-library/jest-dom": "^6.5.0",
+    "@testing-library/react": "^16.0.1",
     "@types/jest": "^29.5.13",
     "@types/node": "^20",
     "@types/react": "^18",
     "@types/react-dom": "^18",
     "cross-env": "^7.0.3",
     "jest": "^29.7.0",
+    "jest-environment-jsdom": "^29.7.0",
+    "npm": "^10.9.0",
```
[the remark this answers is not in the document]
Comment (TJYumul) [reply package.json]: I have already removed it

File: src/shared/components/__test__/landingpage.test.tsx
[the remark this answers is not in the document]
Comment (TJYumul) [reply src/shared/components/__test__/landingpage.test.tsx:1]: I already run the pnpm format

File: src/shared/components/__test__/landingpage.test.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import { cleanup, render, screen, fireEvent } from "@testing-library/react";
+import "@testing-library/jest-dom";
```
[the remark this answers is not in the document]
Comment (TJYumul) [reply src/shared/components/__test__/landingpage.test.tsx]: ctrl s didn't work I don't know why I have the extensions already but I manually organize the imports.

File: src/shared/components/__test__/landingpage.test.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import { Navbar } from "../common/Navbar";
+import { Footer } from "../common/Footer";
+import { SearchBar } from "../home/SearchBar";
+
+afterEach(cleanup);
+
+describe("Navbar", () => {
```
[the remark this answers is not in the document]
Comment (TJYumul) [reply src/shared/components/__test__/landingpage.test.tsx]: I organize the test files

File: src/shared/components/__test__/landingpage.test.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      expect(screen.getByText("NEU Library")).toBeInTheDocument();
+      expect(screen.getByRole("button", {name: /Notifications/i })).toBeInTheDocument();
+      expect(screen.getByRole("button", {name: /Citation Insights/i })).toBeInTheDocument();
+      expect(screen.getByRole("button", {name: /Some Options/i})).toBeInTheDocument();
+      expect(screen.getByRole("button", {name: /Settings/i})).toBeInTheDocument();
+
+  });
+});
+
+describe("SearchBar", () => {
```
[the remark this answers is not in the document]
Comment (TJYumul) [reply src/shared/components/__test__/landingpage.test.tsx]: I organize the test files

File: src/shared/components/__test__/landingpage.test.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      const input = screen.getByPlaceholderText("Search...") as HTMLInputElement;
+      const button = screen.getByRole("button");
+
+      fireEvent.change(input, { target: { value: "Thesis" } });
+      expect(input.value).toBe("Thesis"); 
+
+  });
+});
+
+describe("Footer", () => {
```
[the remark this answers is not in the document]
Comment (TJYumul) [reply src/shared/components/__test__/landingpage.test.tsx]: I organize the test files

File: src/shared/components/common/Navbar.tsx [hunk 6]
```diff
@@ -65,7 +66,7 @@ export const Navbar = () => {
 				</Button>
 			</div>
 			<DropdownMenu>
-				<DropdownMenuTrigger className="rounded-full">
+				<DropdownMenuTrigger className="rounded-full" data-testid="avatar-trigger">
 					<Avatar>
 						<AvatarImage src="https://img.freepik.com/free-psd/3d-rendering-avatar_23-2150833554.jpg?w=740&t=st=1727684407~exp=1727685007~hmac=d94e2a5c2e158613a4c66e0b4623ffba571b35d2be28981616569d772951be60" />
```
[the remark this answers is not in the document]
Comment (TJYumul) [reply src/shared/components/common/Navbar.tsx:71]: I didn't notice the '&'. I added it again. Sorry for the oversight.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (TJYumul) [reply package.json]: I have already removed it

**Review by KuroXI — CHANGES_REQUESTED** [review]
can you rename the component and this test

Component: `NavigationBar`
Test: `navigationBar`

after this I think we can merge this PR

**Review by KuroXI — APPROVED** [review]

**Pull request merged**